### PR TITLE
Allow options to disable user gpg keys configuration from the interface on app.ini

### LINKED ISSUE #29454: Ability to Disable PGP/GPG Keys
### Feature Description

I have migrated from PGP keys to SSH keys and now enforce that only SSH keys are used on my server. I saw pull request #29447 is related to disabling SSH keys (the opposite of what I'm asking for here), and issue #20549 includes GPG keys but no SSH keys.

Is it possible to do the same as what's happening in PR #29447 for PGP/GPG keys?

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by lunny**
Follow #29447
Fix #29454 
Extract from #20549

### CODE REVIEW

**Review by delvh — APPROVED** [review]

File: routers/api/v1/user/gpg_key.go [hunk 1]
```diff
@@ -132,6 +133,11 @@ func GetGPGKey(ctx *context.APIContext) {
 
 // CreateUserGPGKey creates new GPG key to given user by ID.
 func CreateUserGPGKey(ctx *context.APIContext, form api.CreateGPGKeyOption, uid int64) {
+	if setting.Admin.UserDisabledFeatures.Contains(setting.UserFeatureManageGPGKeys) {
+		ctx.NotFound("Not Found", fmt.Errorf("gpg keys setting is not allowed to be visited"))
+		return
+	}
```
Comment (KN4CK3R) [inline routers/api/v1/user/gpg_key.go:139]: Should we move this check into the middleware to keep the logic in the router on topic?

Comment (lunny) [reply routers/api/v1/user/gpg_key.go:139]: I did that on #20549 but @wxiaoguang think it's better to move them into the router function body.

Comment (wxiaoguang) [reply routers/api/v1/user/gpg_key.go:139]: Do you mean this comment? https://github.com/go-gitea/gitea/pull/29275#discussion_r1496895057

At that time:

1. Only one handler needs such check
2. It bloats the "web.go", do we really want to fill more and more function into "web.go"?

For this time: it is still the only way: put checks into `routers/web/user/setting/keys.go`, it is impossible to "move into middleware" for it.


My reviews never means "it must do so", I am always open for better ideas.


So, for this problem: what is the consistent and maintainable approach in your mind?

Comment (KN4CK3R) [reply routers/api/v1/user/gpg_key.go:139]: Something like this for the middleware:
```go
func featureEnabled(feature /*...*/string) func(ctx *context.Context) {
	if setting.Admin.UserDisabledFeatures.Contains(feature) {
		ctx.Error(http.StatusNotFound)
		return
	}
}
```
But that does not work well with the method with the `type` parameter. Repeating the same code multiple times is not really good either.

Comment (wxiaoguang) [reply routers/api/v1/user/gpg_key.go:139]: And by the way, I think my concern for "testing" is still not addressed there https://github.com/go-gitea/gitea/pull/29275#discussion_r1496912355

Comment (wxiaoguang) [reply routers/api/v1/user/gpg_key.go:139]: > But that does not work well with the method with the `type` parameter. Repeating the same code multiple times is not really good either.

There is no consistent way at the moment (if there is no refactoring). You can see that there is no way to use your middleware for the "web.go" now.

----

Update: I didn't mean blocker, and I don't have other ideas besides "refactoring to a clearer approach".

So feel free to use any style people prefer.

Comment (KN4CK3R) [reply routers/api/v1/user/gpg_key.go:139]: Not really no way but we need to use one unpleasant way.

**Review by lafriks — APPROVED** [review]

**Pull request merged**